src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -2987,41 +2988,26 @@ public void withSavedDoc(Command onsaved)
       docUpdateSentinel_.withSavedDoc(onsaved);
    }
 
-
-   // sadly due to Java's memory model we can't modify local variables
-   // inside of a lambda but class variables are protected and
-   // are one of the recommended workarounds
-   private int totalWords = 0;
-   private int selectionWords = 0;
-
    @Handler
    void onWordCount()
    {
-      totalWords = 0;
-      selectionWords = 0;
+      int totalWords = 0;
+      int selectionWords = 0;
 
-      Position pos = docDisplay_.getCursorPosition();
+      Range selectionRange = docDisplay_.getSelectionRange();
       TextFileType fileType = docDisplay_.getFileType();
-      Iterable<Range> wordSource = docDisplay_.getWords(
+      Iterator<Range> wordIter = docDisplay_.getWords(
          fileType.getTokenPredicate(),
          docDisplay_.getFileType().getCharPredicate(),
-         pos,
-         null);
+         Position.create(0, 0),
```

Comment:
Thankfully I was forced to go back and look at this code more thoroughly when I figured out that I could only iterate through the words once. It was only counting words from where the cursor started! This became very obvious as soon as I tested on a small document instead of the larger one I was using before.